sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -827,6 +832,19 @@ public void setValueDeserializer(String valueDeserializer) {
       return toBuilder().setOffsetConsumerConfig(offsetConsumerConfig).build();
     }
 
+    /**
+     * Sets a default timeout to use for API calls to Kafka which are possibly blocking. {@link
+     * KafkaIO.Read} will not wait indefinitely for such calls but cancels them and fail after given
+     * timeout.
+     *
+     * <p>Higher versions of kafka client library supports such API timeout, which is passed in
+     * consumer config as `default.api.timeout.ms`. If timeout is not explicitly set here, then
+     * {@link KafkaIO.Read} will look for `default.api.timeout.ms` parameter in consumer config.
+     */
+    public Read<K, V> withDefaultKafkaApiTimeout(Duration timeout) {
```

Comment:
Instead of exposing it into user API as a separate call, can we just leverage `default.api.timeout.ms` config option if it's present? If it was not set by default (old version of Kafka client, for example) then we can ask user to set explicitly with `withConsumerConfigUpdates()`? And finally, it was not set in any case, just use default value as it was before.

Reply:
Yes, I did think of that. I was not sure if to choose that path - less transparent but easier to maintain. I suppose this can be kept at the level of "advanced configuration" which needs code familiarity. I will make these changes and push an update

Reply:
Thanks! Please, add a comment about that into KafkaIO class javadoc to make it clear for users.